staging/src/k8s.io/client-go/tools/remotecommand/fallback.go

```diff
@@ -0,0 +1,66 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package remotecommand
+
+import (
+	"context"
+	"net/url"
+
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/client-go/transport/websocket"
+)
+
+var _ Executor = &fallbackExecutor{}
+
+type fallbackExecutor struct {
+	primary   Executor
+	secondary Executor
+}
+
+// NewFallbackExecutor creates an Executor that first attempts to use the
+// WebSocketExecutor, falling back to the legacy SPDYExecutor if the initial
+// websocket "StreamWithContext" call fails.
+func NewFallbackExecutor(config *restclient.Config, method string, url *url.URL) (Executor, error) {
```

Comment:
rather than baking in the construction/knowledge of websocket and spdy, model this as

```
NewFallbackExecutor(primary, secondary Executor, shouldFallback func(error) bool) (Executor, error)
```

then the caller can pass in whatever primary/secondary executor they want, and whatever condition they want to decide whether to fallback

Reply:
Fixed.